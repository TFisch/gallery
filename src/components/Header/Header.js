Vue.component('header', {
  template: '<li>This is a todo</li>'
});
