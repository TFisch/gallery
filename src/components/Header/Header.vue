<template>
<div class="header">
   <h1 class="text">{{text}}</h1>
</div>
</template>

<script>
export default {
  name: 'Header',
  data() {
    return {
      text: 'Japanese Paintings'
    };
  }
};
</script>

<style scoped>
.header {
  background-color: lime;
  border-bottom: 1px solid black;
  height: 3em;
  margin: auto;
}

.text {
  color: brown;
  margin: auto;
}
</style>
