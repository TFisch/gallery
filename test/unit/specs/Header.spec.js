import { shallowMount } from '@vue/test-utils';
import Vue from 'vue';
import Header from './../../../src/components/Header/Header';

describe('Header.vue', () => {
  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallowMount(Header, {});
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render the page title', () => {
    const text = 'Japanese Paintings';
    wrapper = shallowMount(Header, {});
    expect(wrapper.text()).toBe(text);
  });
});
