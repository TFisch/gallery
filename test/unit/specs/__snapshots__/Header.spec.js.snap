// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue should match the snapshot 1`] = `"<div class=\\"header\\"><h1 class=\\"text\\">Japanese Paintings</h1></div>"`;
